src/Server/Players/PlayersRepository.fs

```diff
@@ -44,20 +44,38 @@ module Database =
     }
 
   let getOrCreateByName connectionString (PlayerName name) : Task<Result<Player, exn>> =
-    task {
-      let! existingPlayer = getByName connectionString (PlayerName name)
-      match existingPlayer with
-      | Error ex -> return Error ex
-      | Ok (Some player) -> return Ok player
-      | Ok None ->
+    // Helper functions
+    let condExec f m = task {
```

Comment:
Would it be worth using the full name here?

Reply:
I don't understand what you mean ? What full name ?

Reply:
Something like contitionalExecute rather than condExec

Reply:
Sure thing...

Reply:
Also could we please have names for the parameters rather than just f and m?

Reply:
What about `function` and `result` ?

Reply:
BTW, since I don't know you coding style, please feel free to make all the changes you would like after the merge. Make it yours. ;) hahaha

Reply:
something like this:
```fsharp
let condExec taskFunction previousResult = task {
    match previousResult with
    | Error ex -> return Error ex
    | Ok (Some player) -> return Ok (Some player)
    | _ -> return! taskFunction()
}
```

Reply:
I like these. Thank you - very descriptive.